From: Lino Sanfilippo <LinoSanfilippo@gmx.de>
Date: Sat, 18 Feb 2017 12:19:41 +0100
Subject: net: ena: remove superfluous check in ena_remove()
Patch-mainline: v4.11-rc1
Git-commit: 90a6c997bd8bb836809eda5efb6406d8c58c0924
References: bsc#1027968

The check in ena_remove() for the pci driver data not being NULL is not
needed, since it is always set in the probe() function. Remove the
superfluous check.

Signed-off-by: Lino Sanfilippo <LinoSanfilippo@gmx.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/amazon/ena/ena_netdev.c |    6 ------
 1 file changed, 6 deletions(-)

--- a/drivers/net/ethernet/amazon/ena/ena_netdev.c
+++ b/drivers/net/ethernet/amazon/ena/ena_netdev.c
@@ -3163,12 +3163,6 @@ static void ena_remove(struct pci_dev *p
 	struct ena_com_dev *ena_dev;
 	struct net_device *netdev;
 
-	if (!adapter)
-		/* This device didn't load properly and it's resources
-		 * already released, nothing to do
-		 */
-		return;
-
 	ena_dev = adapter->ena_dev;
 	netdev = adapter->netdev;
 
